File: Baekjoon/S5/C#/S5_소수 찾기.cs
https://www.acmicpc.net/problem/1978

using System;

namespace Baekjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            string[] arrStr = Console.ReadLine().Split(' ');
            int[] arr = Array.ConvertAll(arrStr, int.Parse);

            int ctr = 0;
            for (int i = 0; i < n; i++)
            {
                if (arr[i] < 2) continue;
                for (int j = 2; j <= Math.Sqrt(arr[i]); j++)
                {
                    if (arr[i] % j == 0)
                    {
                        arr[i] = 0;
                        break;
                    }
                }
                if (arr[i] != 0)
                    ctr++;
            }

            Console.WriteLine(ctr);
        }
    }
}

File: Baekjoon/S5/JAVA/S5_소수 찾기.java
//https://www.acmicpc.net/problem/1978

import java.io.*;

public class Main{
    public static void main(String args[]) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        String[] arr = br.readLine().split(" ");
        
        int cnt = 0;
        for(int i = 0; i < n; i++){
            int num = Integer.parseInt(arr[i]);
            
            if(check(num))
                cnt++;
        }
        
        System.out.println(cnt);
        br.close();
    }
    
    static boolean check(int n){
        if(n == 1)
            return false;
            
        for(int i = 2; i <= Math.sqrt(n); i++){
            if(n%i == 0)
                return false;
        }
        
        return true;
    }
}